crates/primitives/src/chain/spec.rs

```diff
@@ -1518,23 +1517,36 @@ pub struct DisplayHardforks {
 
 impl Display for DisplayHardforks {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        writeln!(f, "Pre-merge hard forks (block based):")?;
-        for fork in self.pre_merge.iter() {
-            writeln!(f, "- {fork}")?;
+        fn format(
+            header: &str,
+            forks: &Vec<DisplayFork>,
+            next_is_empty: bool,
+            f: &mut Formatter<'_>,
+        ) -> std::fmt::Result {
+            writeln!(f, "{}:", header)?;
+            for (index, fork) in forks.iter().enumerate() {
+                if next_is_empty && index == forks.len() - 1 {
```

Comment:
style: I believe can be solved with:

```rust
let mut iter = forks.iter().enumerate().peekable();
while let Some((index, fork)) = iter.next() {
   write!(f, "- {}", fork)?;
   if iter.peek().is_some() && !next_is_empty {
      writeln!(f)
   }
}
```

Reply:
thanks, this looks more clear and functionable